Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm

```diff
@@ -44,35 +44,55 @@ @implementation _WKWebExtension
 
 WK_OBJECT_DEALLOC_IMPL_ON_MAIN_THREAD(_WKWebExtension, WebExtension, _webExtension);
 
-+ (instancetype)extensionWithAppExtensionBundle:(NSBundle *)appExtensionBundle
++ (void)extensionWithAppExtensionBundle:(NSBundle *)appExtensionBundle completionHandler:(void (^)(_WKWebExtension *extension, NSError *error))completionHandler
 {
     NSParameterAssert([appExtensionBundle isKindOfClass:NSBundle.class]);
 
-    NSError * __autoreleasing internalError;
-    auto result = WebKit::WebExtension::create(appExtensionBundle, &internalError);
+    // FIXME: <https://webkit.org/b/276194> Make the WebExtension class load data on a background thread.
+    // Use an async dispatch in the meantime to prevent clients from expecting synchronous results.
 
-    if (internalError)
-        return nil;
+    dispatch_async(dispatch_get_main_queue(), ^{
+        NSError * __autoreleasing error;
+        Ref result = WebKit::WebExtension::create(appExtensionBundle, &error);
```

Comment:
I think it would be a little nicer to make WebExtension::create return an Expected&lt;Ref&lt;WebExtension&gt;, RetainPtr&lt;NSError&gt;&gt; but that's definitely unrelated to this PR.

Reply:
Agreed! I'll note that for another change.